Purchases/RCBackend.m

```diff
@@ -393,9 +413,69 @@ - (void)createAliasForAppUserID:(NSString *)appUserID
                                        @"new_app_user_id": newAppUserID
                                }
                             headers:self.headers
-                  completionHandler:^(NSInteger status, NSDictionary *response, NSError *error) {
+                  completionHandler:^(NSInteger status, NSDictionary *_Nullable response, NSError *_Nullable error) {
                       [self handle:status withResponse:response error:error errorHandler:completion];
                   }];
 }
 
+
+- (void)postOfferForSigning:(NSString *)offerIdentifier
+      withProductIdentifier:(NSString *)productIdentifier
+          subscriptionGroup:(NSString *)subscriptionGroup
+                       data:(NSData *)data
+                  appUserID:(NSString *)appUserID
+                 completion:(RCOfferSigningResponseHandler)completion
+{
+    NSString *fetchToken = [data base64EncodedStringWithOptions:0];
+    [self.httpClient performRequest:@"POST" path:@"/offers"
+                               body:@{
+                                       @"app_user_id": appUserID,
+                                       @"fetch_token": fetchToken,
+                                       @"generate_offers": @[@{
+                                               @"offer_id": offerIdentifier,
+                                               @"product_id": productIdentifier,
+                                               @"subscription_group": subscriptionGroup
+                                       }],
```

Comment:
Having the body at the call site is getting a little hairy to read. Maybe better to pull it out to a var.